scripts/datastore_upgrade.py

```diff
@@ -0,0 +1,330 @@
+import json
+import logging
+import os
+import subprocess
+import sys
+import time
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../lib'))
+import appscale_info
+from constants import CONTROLLER_SERVICE
+from constants import APPSCALE_HOME
+import monit_interface
+import monit_app_configuration
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../AppDB'))
+import appscale_datastore_batch
+import datastore_server
+from dbconstants import *
+from zkappscale import zktransaction as zk
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "../AppServer"))
+from google.appengine.datastore import entity_pb
+
+# The location on the local filesystem where Cassandra is installed.
+CASSANDRA_DIR = "/opt/cassandra"
+
+# The location of the Cassandra binary on the local filesystem.
+CASSANDRA_EXECUTABLE = CASSANDRA_DIR + "/cassandra/bin/cassandra"
+
+PID_FILE = "/var/appscale/appscale-cassandra.pid"
+
+# The default port to connect to Cassandra.
+CASSANDRA_PORT = 9999
+
+# The default port to connect to ZooKeeper.
+ZOOKEEPER_PORT = 2181
+
+# Default ZooKeeper port formatted with a colon.
+ZK_PORT_WITH_COLON = ":" + str(ZOOKEEPER_PORT)
+
+# The location on the local filesystem where we should store ZooKeeper data.
+ZK_DATA_LOCATION = "/opt/appscale/zookeeper"
+
+# The maximum number of entities to be returned from the query.
+_MAX_ENTITIES = 1000000
+
+def ensure_app_is_not_running():
+  """ Ensures AppScale is not running as this is an offline script. """
+  logging.info("Ensure AppScale is not currently running...")
+  appscale_running = subprocess.call(['service', CONTROLLER_SERVICE, 'status']) == 0
+  if appscale_running:
+    logging.info("AppScale is running, please shut it down and try again.")
+    sys.exit(1)
+
+def start_cassandra():
+  """ Creates a monit configuration file and prompts Monit to start Cassandra. """
+  logging.info("Starting Cassandra...")
+  watch = "cassandra"
+  start_cmd = CASSANDRA_EXECUTABLE + " start -p " + PID_FILE
+  stop_cmd = "/usr/bin/python2 " + APPSCALE_HOME + "/scripts/stop_service.py java cassandra"
+  monit_app_configuration.create_config_file(watch, start_cmd, stop_cmd,
+    ports=[CASSANDRA_PORT], upgrade_flag=True, match_cmd=CASSANDRA_DIR)
+  if not monit_interface.start(watch):
+    logging.info("Unable to start Cassandra.")
+    sys.exit(1)
+  else:
+    logging.info("Successfully started Cassandra.")
+
+def start_zookeeper():
+  """ Creates a monit configuration file and prompts Monit to start ZooKeeper. """
+  logging.info("Starting ZooKeeper...")
+  os.system("rm -rfv /var/lib/zookeeper")
```

Comment:
Look [here](https://docs.python.org/2/library/os.html) for removing files, creating dirs, etc.

As Chris mentioned, if a function is not available for what you want to do it's better to use subprocess.